src/project/internal/notationproject.h

```diff
@@ -90,6 +103,9 @@ class NotationProject : public INotationProject
     notation::MasterNotationPtr m_masterNotation = nullptr;
     ProjectAudioSettingsPtr m_projectAudioSettings = nullptr;
     ProjectViewSettingsPtr m_viewSettings = nullptr;
+
+    SaveLocation m_saveLocation = SaveLocation::makeInvalid();
+    QFileInfo m_cachedFileInfo;
```

Comment:
I think, EngravingProject can provide this information directly without any caches here because it already stores a pointer to IFileInfoProvider